package.json

```diff
@@ -135,7 +135,49 @@
         "language": "solidity",
         "path": "./snippets/solidity.json"
       }
-    ]
+    ],
+    "taskDefinitions": [
+      {
+        "type": "hardhat",
+        "required": [
+          "task"
+        ],
+        "properties": {
+          "task": {
+            "type": "string",
+            "description": "%taskdef.task%"
+          }
+        },
+        "when": "shellExecutionSupported"
+      }
+    ],
+    "commands": [
+      {
+        "command": "hardhat.solidity.compile",
+        "title": "Hardhat: Compile project"
+      },
+      {
+        "command": "hardhat.solidity.clean",
+        "title": "Hardhat: Clear cache and artifacts "
+      },
+      {
+        "command": "hardhat.solidity.flattenCurrentFile",
+        "title": "Hardhat: Flatten this file and their dependencies",
+        "enablement": "editorLangId == 'solidity'"
+      }
+    ],
+    "menus": {
+      "editor/context": [
+        {
+          "when": "editorLangId == 'solidity'",
+          "command": "hardhat.solidity.compile"
+        },
+        {
+          "when": "editorLangId == 'solidity'",
+          "command": "hardhat.solidity.flattenCurrentFile"
```

Comment:
This is working on the file on disk rather than the in-memory open document.

Reply:
Hmm good point. Should we prompt the user to save the document before running the flattening ?